### PR TITLE
added checks for uninstall and upgrade

### LINKED ISSUE #816: CLI improvements
Could `kusk cluster uninstall` check if Kusk is already installed and then proceeds to delete it. Otherwise, running it on a cluster that doesn't have Kusk blows up the error message

Similar with `kusk cluster upgrade`. Also, could we instead of installing the new version, check if it's already the latest and make a message clear that we didn't update? 

`kusk dashboard`
- look for an available port if 8080 is not there.. see Testkube's example https://github.com/kubeshop/testkube/pull/1773
- if Kusk is not installed in the cluster return a suggestion to install Kusk

### PR DESCRIPTION
**Pull request by jasmingacic**
Signed-off-by: jasmingacic <jasmin.gacic@gmail.com>

Partially fixes #816

### CODE REVIEW

File: cmd/kusk/internal/utils/githubclient_test.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package utils
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestGetTags(t *testing.T) {
```
Comment (mbana) [inline cmd/kusk/internal/utils/githubclient_test.go:8]: Why not just use the `github.com/stretchr/testify/assert` package?

File: cmd/kusk/cmd/upgrade.go [hunk 2]
```diff
@@ -172,3 +187,22 @@ func init() {
 	upgradeCmd.Flags().StringVar(&releaseNamespace, "namespace", "kusk-system", "namespace to upgrade in")
 	upgradeCmd.Flags().BoolVar(&installOnUpgrade, "install", false, "install components if not installed")
 }
+
+func getVersions(component, container string, deployment appsv1.Deployment) (latest string, current string) {
```
Comment (mbana) [inline cmd/kusk/cmd/upgrade.go:191]: Shouldn't you return the `error`s that could potentially occur in this function, or at least report them?

File: cmd/kusk/cmd/upgrade.go [hunk 3]
```diff
@@ -78,26 +80,34 @@ var upgradeCmd = &cobra.Command{
 			reportError(err)
 			return err
 		}
+
+		if len(deployments.Items) == 0 {
+			kuskui.PrintInfo("Kusk is not installed on the cluster")
+			os.Exit(0)
```
Comment (mbana) [inline cmd/kusk/cmd/upgrade.go:86]: Missing `reportError(err)` or intentional?

File: cmd/kusk/cmd/upgrade.go [hunk 4]
```diff
@@ -115,49 +125,54 @@ var upgradeCmd = &cobra.Command{
...
-					kuskui.PrintError("❌ failed upgrading API Server", err.Error())
-					reportError(err)
-					return err
-				}
-				if err := utils.WaitForPodsReady(cmd.Context(), c, namespace, "kusk-gateway-api", time.Duration(5*time.Minute), "instance"); err != nil {
-					kuskui.PrintError("❌ failed upgrading API Server", err.Error())
-					reportError(err)
-					return err
-				}
+				if !utils.IsUptodate(getVersions(deployment.Name, "kusk-gateway-api", deployment)) {
```
Comment (mbana) [inline cmd/kusk/cmd/upgrade.go:128]: Shouldn't these be constants? `"kusk-gateway-api"` is repeated in the line above in the `case` statement.

File: cmd/kusk/cmd/upgrade.go [hunk 5]
```diff
@@ -115,49 +125,54 @@ var upgradeCmd = &cobra.Command{
...
-					kuskui.PrintError("❌ failed upgrading Dashboard", err.Error())
-					reportError(err)
-					return err
-				}
-				if err := utils.WaitForPodsReady(cmd.Context(), c, namespace, "kusk-gateway-dashboard", time.Duration(5*time.Minute), "instance"); err != nil {
-					kuskui.PrintError("❌ failed upgrading Dashboard", err.Error())
-					reportError(err)
-					return err
-				}
+				if !utils.IsUptodate(getVersions("kusk-gateway-dashboard", "kusk-gateway-dashboard", deployment)) {
```
Comment (mbana) [inline cmd/kusk/cmd/upgrade.go:149]: Shouldn't these be constants? `"kusk-gateway-dashboard"` is repeated in the line above in the `case` statement.

**Review by mbana — COMMENTED** [review]
Left some remarks.

File: cmd/kusk/cmd/uninstall.go [hunk 6]
```diff
@@ -54,68 +56,86 @@ var uninstallCmd = &cobra.Command{
...
 				return err
 			}
 
-			c, err := utils.GetK8sClient()
-			if err != nil {
+			kuskui.PrintStart("Checking if kusk is already installed...")
+
+			kuskNamespace := &corev1.Namespace{}
+			if err := c.Get(cmd.Context(), client.ObjectKey{Name: "kusk-system"}, kuskNamespace); err != nil {
+				if err.Error() == `namespaces "kusk-system" not found` {
```
[the remark this answers is not in the document]
Comment (jasmingacic) [reply cmd/kusk/cmd/uninstall.go:69]: it isn't an error just a print out that Kusk isn't installed on the cluster. 
So yes it is intentional

File: cmd/kusk/internal/utils/githubclient_test.go (hunk 1 above)
Comment (jasmingacic) [reply cmd/kusk/internal/utils/githubclient_test.go:8]: It is important as long it isn't not failing

**Review by mbana — APPROVED** [review]
LGTM.

**Pull request merged**